### DISCUSSION TITLE
Global Decode Quality Question - Just Making Sure

### DISCUSSION THREAD

**Discussion by tangierc** [latenitefilms/BRAWToolbox#143] [General]
Hey folks. Perhaps I am overthinking something and briefly confused, but it can't hurt to ask to be absolutely sure. Does the global decode quality setting in the inspector for a clip apply to wherever that clip exists or for all BRAW Toolbox created clips in the open library? I ask because I noticed that when I went into the inspector and selected Global settings, this was set to Force Full Decode Quality. At the same time my clip decode quality shown in the parameter metadata just below the global setting was set to eighth. I just want to insure that the images I see and export are full quality.  I also opened that workflow extension to make sure it was set to Force Full Decode Quality. 
![Screenshot 2023-06-28 at 4 44 20 PM](https://github.com/latenitefilms/BRAWToolbox/assets/9577806/f4bb8623-66f4-4ff4-93db-64a9b50fdea3)
![Screenshot 2023-06-28 at 4 44 03 PM](https://github.com/latenitefilms/BRAWToolbox/assets/9577806/64530bc0-2681-476d-b7ef-093ba459e68f)

**Comment by latenitefilms** [comment]
If you have **Force Full Decode Quality** ticked, then this basically tells the BRAW Toolbox renderer to render EVERYTHING at Full Decode Quality across the board. If this is ticked EVERYTHING will be rendered at Full Decode Quality.

**Reply by MathewWelsh** [reply]
I'm working on a project shot on Black Magic camera at 6K. Two days of filming for a finished 10 minute video. 100s of clips and content.

My Mac is a relatively powerful, but older, intel system. I prefer to work with proxies during editing for smoother playback, etc. But I understand that with BRAW Toolbox, the suggested workflow is to take advantage of the decode quality settings. I have spent a couple of hours now trying to figure out how to down-res all the clips imported into FCP using BRAW Toolbox. 

(I've asked Perplexity, watched YouTube videos, read around your GitHub issues and discussion board, and I've read your webpage on BRAW Toolbox - https://brawtoolbox.io/.

I see here https://brawtoolbox.io/ > "There are automatic Decode Quality options for both HD and UltraHD projects, and you can also work at a low quality Decode Quality, then apply a Global Setting to force Full Quality prior to export. BRAW Toolbox gives you complete freedom and flexibility."

I imported my BRAW files into FCP at the Decode Quality setting of Automatic UltraHD,  for all clips. Now, can I down-res those with a Global Setting that forces Quarter Decode Quality? I've tried this by opening one clip to reveal the BRAW controls in the FCP Inspector. I've clicked on Global Settings, then "Force quarter decode quality." But that doesn't even change the decode quality for the clip that I've selected, let alone for all clips in my project. See screenshot to see what I mean.

Can you please share with us the workflow for down-rezing to a lower decode quality for editing with a "proxy-like" workflow, then up-rezing for final export? THX.

<img width="1371" alt="Decode quality conundrum" src="https://github.com/user-attachments/assets/0cf6dacb-2a6b-4d5c-b118-88030a02a068">

**Reply by latenitefilms** [reply]
Unless there's some weird bug, the Force Decode Quality SHOULD be working. Is it possible you've already rendered that section of the timeline?

**Reply by MathewWelsh** [reply]
Thanks very much for your reply! I'm still editing away here, and I guess it's morning in Australia...
I am not savvy enough to figure out if there's a weird bug. 
Not sure about the render question. In my FCP settings I have had background render checked the whole time. I just changed that in settings so that background render is unchecked. So it's likely that sections of the timeline have rendered as I've been working. If that's the case, what do you suggest? Thanks very much.

**Reply by latenitefilms** [reply]
If you had background render checked, then Final Cut Pro would have been constantly rendering all the BRAW Toolbox clips, and essentially "baking in" the BRAW footage, so when you watch back the rendered clips, the decode quality would have no impact - as Final Cut Pro would have just been playing back the ProRes renders from disk. Now that you have background rendering disabled, Final Cut Pro won't be rendering anything in the background, so you should notice the decode quality on new clips on the timeline - i.e. if you force to Eighth Decode quality, it'll look worse quality than full decode quality. These settings also affect exporting, so make sure you force full decode quality prior to exporting.

Hope this helps! Any questions let me know.

**Reply by MathewWelsh** [reply]
okay, thanks, I'm slowly understanding how your BRAW Toolbox works, and how to think about it.
Question: how can I undo the rendering so that I can have quarter size files on my timeline as I edit?
in other words, How do I backtrack to a point where I can reset the Decode Quality to 1/4, and have it apply across all clips in my timeline, etc. Thanks for your advice here. This might seem like a basic thing most editors would know, but I'm mainly a producer who pretends to be an editor, doing the rough cuts as a writing tool. Thanks.

**Reply by latenitefilms** [reply]
Everything that's already rendered should just be playing back the ProRes files on disk, so they should be fast anyway?

You can delete all your render files via the **File > Delete Generated Library Files...** buttons:

<img width="545" alt="image" src="https://github.com/user-attachments/assets/21ef755a-438b-4c8a-9094-8d7f4efa2185">

**Reply by MathewWelsh** [reply]
Thanks very much, I'll give this a try in the morning. Much appreciated!

**Reply by MathewWelsh** [reply]
no luck.
I turned off background render in my settings. 

I deleted generated clips, then I restarted FCP. Then I went into a number of clips to look at the Global Settings. When I opened the BRAW tools in the inspector, it showed that Force Quarter Decode Quality was checked, but on all clips, the parameters below show that it's still on Automatic HD. See screenshot.

Is there still something else I'm not doing correctly?

<img width="1343" alt="Force Quarter Decode bug?" src="https://github.com/user-attachments/assets/9c8c638a-310c-4a0b-83c4-97f7a73a1c93">

**Reply by latenitefilms** [reply]
Pressing the "Force" tickboxes doesn't actually change the Decode Quality dropdown in the Final Cut Pro Inspector (as you may have different clips with different decode qualities) - it just affects how we render things, so you should see a visible difference in the Viewer. Make sure you have your Viewer in "Best Quality" mode. How dramatic the changes look really depends on the resolution of your original BRAW footage, and what your timeline resolution is - ie 8K clips in a HD timeline will look different than 8K clips in an 8K timeline, etc. "Automatic (HD)", tries to determine the optimal decode quality for your BRAW footage for HD output.

**Reply by MathewWelsh** [reply]
Okay, bear with me for a clarifying question: I'm trying to understand the difference between working with proxies (using normal prores footage in a normal FCP workflow) and using Decode Quality with BRAW through your Toolbox.

Are you saying that the Global setting just affects exports and renders?

So if I'm seeing chugging in the system, as FCP slugs through 6K footage as I play my timeline in the viewer, this is because the footage is still 6K. It's ONLY when I go to export that it will render and export at 1/4 size (1/4 of 6K)?

Here's a question: Now that I've deleted the generated files, and turned off the background render, and set the global settings to Force Quarter Decode, should I turn back on the background render, so that as FCP renders while I work, it's rendering in 1/4 size, and then as I work the edit, FCP will always be referencing those new 1/4 size files, and it will have an easier time wtih playback, thus SORT OF emulating the experience of working with proxies?

Have I got that right?

**Comment by tangierc** [comment]
Thanks Chris for clarifying. Had a bit of a panic as I have been delivering files to the color team, but have had Force Full Decode Quality on for a while. In my mind perhaps I was thinking that by changing the global setting, the metadata setting would dynamically update to match the global. Since I saw something different, it gave me pause to question what was actually happening.

P.S. 

Spoke about you on RichardTaylor TVs Final Cut Pro 12th celebration recently. Looking forward to what you come up with for Sonoma and BRAW with the new media framework.

**Reply by latenitefilms** [reply]
Unfortunately I've been able to get zero information out of Apple about the new MediaExtension Framework. It requires a special entitlement, but there's zero information on how to get the entitlement:

<img width="974" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/4426fc74-b75e-4d40-a0e3-8c76b5d506c0">
